packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -111,6 +113,159 @@ function validateDataStoreStateInSummary(
 	);
 }
 
+const tombstoneTimeoutMs = 200;
+const sweepGracePeriodMs = 0; // Skip Tombstone, these tests focus on Sweep
+const sweepTimeoutMs = tombstoneTimeoutMs + sweepGracePeriodMs;
+const gcOptions: IGCRuntimeOptions = {
+	inactiveTimeoutMs: 0,
+	enableGCSweep: true,
+	sweepGracePeriodMs,
+};
+const configProvider = createTestConfigProvider();
+const testContainerConfig: ITestContainerConfig = {
+	runtimeOptions: {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				state: "disabled",
+			},
+		},
+		gcOptions,
+	},
+	loaderProps: { configProvider },
+};
+
+let provider: ITestObjectProvider;
+
+async function loadContainer(
+	summaryVersion: string,
+	config: ITestContainerConfig = testContainerConfig,
+) {
+	return provider.loadTestContainer(config, {
+		[LoaderHeader.version]: summaryVersion,
+	});
+}
+
+const loadSummarizer = async (container: IContainer, summaryVersion?: string) => {
+	return createSummarizer(
+		provider,
+		container,
+		{
+			runtimeOptions: { gcOptions },
+			loaderProps: { configProvider },
+			forceUseCreateVersion: true, // To simulate the summarizer running on the created container
```

Comment:
@agarwal-navin - do you have a strong opinion on this v. the alternative approach to have the summarizer circumvent the provider and always be current?

Reply:
Currently, this is only required for summarizer and for one specific test so my preference would be to not add to the generic test infra. If this becomes more common then we can consider doing this.
The TestObjectProvider also exposes `apis` which can be used to create current compat version of container runtime / data store runtime, etc. You can try using that too.

Reply:
I tried it, it just lead to more problems.  Navin and I agreed to leave it as-is.